src/components/CommentCard.jsx

```diff
@@ -1,18 +1,45 @@
-import moment from "moment"
+import { deleteCommentFromArticle } from "./api"
+import { useParams } from "react-router"
+import { useState, useContext } from "react"
+import { UserLoginContext } from "../contexts/UserLogin"
 
-function CommentCard({comment}) {
+function CommentCard({comment, setRefresh, isLoading, setIsLoading}) {
+
+    const {comment_id} = comment
+    const [isDeleteButtonDisabled, setIsDeleteButtonDisabled] = useState(false)
+    const { loggedUser } = useContext(UserLoginContext)
+    
 
     const dateAndTime = comment.created_at.split("T")
     const time = dateAndTime[1].split(":")
     const date = dateAndTime[0].split("-")
+
+    function disableDeleteButton() {
+        setIsDeleteButtonDisabled(true)
+    }
+
+    function handleDelete(event) {
+        event.preventDefault()
+        disableDeleteButton()
+        setIsLoading(true)
+            deleteCommentFromArticle(comment.comment_id)
```

Comment:
We're not optimistically rendering here! Please review the notes - we should be setting the state with the comment removed at the same time as triggering the db request.

Reply:
I see what you're doing here with setRefresh but it's not the correct use of state. Rather than using this to trigger a re-render, you should be optimistically rendering a locally update list of the comments with the deleted comment removed. Then the next time someone refreshes/navigates to the page, the list provided will be the reflection of the db update.